src/utils/globals.ts

```diff
@@ -4,9 +4,11 @@
 export const ArrayProto = Array.prototype
 export const nativeForEach = ArrayProto.forEach
 export const nativeIndexOf = ArrayProto.indexOf
-export const win: Window & typeof globalThis = typeof window !== 'undefined' ? window : ({} as typeof window)
-const navigator = win.navigator || { userAgent: '' }
-const document = win.document || {}
-const userAgent = navigator.userAgent
+// eslint-disable-next-line no-restricted-globals
+export const win: (Window & typeof globalThis) | undefined = typeof window !== 'undefined' ? window : undefined
+const navigator = win?.navigator
+export const document = win?.document
+export const userAgent = navigator?.userAgent
+export const assignableWindow: Window & typeof globalThis & Record<string, any> = win ?? ({} as any)
```

Comment:
The `Record<string, any>` makes me wonder if we shouldn't make that an explicit type of the things we have added to window... but that can be a follow-up no reason to add more here